src/main/java/com/sivalabs/bookstore/catalog/domain/PagedResult.java

```diff
@@ -22,6 +22,7 @@
     @JsonProperty("hasPrevious")
     private boolean hasPrevious;
 
+    // TODO: this can be made private constructor in favor of factory method `fromPage`
```

Comment:
Can use Java Record for simplicity

Reply:
Done ✔️ 
Nice suggestion @rajadilipkolli , I wasn't aware that jackson started supporting `record`  🍍

Reply:
> Done ✔️ Nice suggestion @rajadilipkolli , I wasn't aware that jackson started supporting `record` 🍍

I'm Glad that you have learnt something new